source/features/collapse-markdown-sections.ts

```diff
@@ -3,12 +3,18 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
-function onHeadingClick({delegateTarget: sectionHeading}: delegate.Event<MouseEvent, HTMLElement>): void {
+function onHeadingClick(event: delegate.Event<MouseEvent, HTMLElement>): void {
 	// Don't toggle the section if the title text is being selected instead of clicked
 	if (document.getSelection()?.type === 'Range') {
 		return;
 	}
 
+	// Don't toggle the section if a link in the heading is clicked (either the content or the anchor)
+	if ((event.target as HTMLElement).closest('h1, h2, h3, h4, h5, h6, a')!.tagName === 'A') {
```

Comment:
```suggestion
	if ((event.target as HTMLElement).closest('a')) {
```

It doesn't matter which one is the closest, right? If it's a link, then it should be ignored

Reply:
Yes but what if one of the ancestors is a link outside the heading element?